packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2256,9 +2216,11 @@ def check_transaction(self, transaction: Transaction) -> None:
 
         :param transaction: the transaction.
         """
+
+        payload_type = type(transaction.payload)
         if (
             self.is_termination_set
-            and transaction.payload.transaction_type == self.background_round_tx_type
+            and payload_type is cast(AppState, self.background_round_cls).payload_class
```

Comment:
now this is fail safe. If termination set, `background_round_cls` is set. The `payload_class` can technically be `None` (though supposedly not in practice), hence we do the unidiomatic type-checking using `type()`